### ISSUE TITLE
Custom formats

### ISSUE THREAD

**Issue by tujoworker** [Sqooba/mobx-react-intl#10]
Hoi Christopher!

I wold love to have things just a little more flexible, so we could add custom props like formats. Have a look on the code above:

```javascript
//source https://github.com/yahoo/react-intl/issues/539

const formats = {
    number: {
        USD: {
            style: 'currency',
            currency: 'USD'
        }
    }
};

<IntlProvider
    locale='en'
    messages={{
        price: 'This product costs {price, number, USD}'
    }}
    formats={formats}
    defaultFormats={formats}
>
    <FormattedMessage id='price' values={{price: 1000}}/>
    <FormattedNumber value={1000} format='USD'/>
</IntlProvider>
```

For sure, I can use e.g. ```jsx <FormattedNumber value={1000} style="currency" currency="CHF" />``` for now.

**Comment by ChrisJamesC** [comment]
Sounds good! We could just pass all parameters to the IntlProvider through the spread operator.

**Comment by tujoworker** [comment]
Nice, this fix works fine! By the way, how do You setting up the workflow to test/change this package?

**Comment by ChrisJamesC** [comment]
Thanks, I use `yarn link`

**Comment by ChrisJamesC** [comment]
Fixed in #11
